spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/BootApplicationConfigurationMetadataResolver.java

```diff
@@ -106,29 +97,18 @@ public boolean supports(Resource app) {
 	 *            but directories are supported as well
 	 */
 	public List<ConfigurationMetadataProperty> listProperties(Resource app, boolean exhaustive) {
-		File moduleFile = null;
 		try {
-			moduleFile = app.getFile();
-		}
-		catch (IOException e) {
-			return Collections.emptyList();
-		}
-		Archive archive = null;
-		try {
-			archive = moduleFile.isDirectory() ? new ExplodedArchive(moduleFile) : new JarFileArchive(moduleFile);
+			Archive archive = resolveAsArchive(app);
 			return listProperties(archive, exhaustive);
 		}
 		catch (IOException e) {
-			throw new RuntimeException("");
+			throw new RuntimeException("Failed to list properties for " + app, e);
 		}
 	}
 
 	public List<ConfigurationMetadataProperty> listProperties(Archive archive, boolean exhaustive) {
-		ClassLoader moduleClassLoader = null;
-		try {
-
+		try (URLClassLoader moduleClassLoader = new BootClassLoaderFactory(archive, parent).createClassLoader()) {
```

Comment:
https://github.com/spring-cloud/spring-cloud-dataflow/pull/872/files/349131983ca59b01e4474481fb2c56ba3140dcfb#diff-0f08760dc293294b1be2538fae1a613fR151 for example. Needed to see the enum that is in `src/test/java` there